rfcs/000-third-party-software-installation.md

```diff
@@ -0,0 +1,109 @@
+- Topic Name: `third-party-software-installation`
+- Start Date: 2022-03-22
+- RFC PR: [vlang/rfcs#0000](https://github.com/vlang/rfcs/pull/22)
+  <!-- - V Issue: [vlang/v#00000](https://github.com/vlang/v/issues/00000) -->
+
+# Summary
+
+Allow installation of third-party V software.
+This goes beyond just placing the source code in `VMODULES`.
+It allows building the software from source locally and adding it to path for the user to uses from anywhere.
+
+Additionally, we propose the use of `~/.v` and the relevant subfolders (like `~/.v/modules`, `~/.v/bin`) to store the source code, built binaries, etc.
+
+# Motivation
+
+Currently, V only allows downloading the source code for V modules in the `VMODULES` directory through the `v install` command.
+This is good for making modules globally available. There is currently no streamlined way of _installing_ or _fetching_ or _obtaining_ third-party software written in V.
+
+# Guide-level explanation
+
+Take [_klonol_](https://github.com/hungrybluedev/klonol) for example.
+It is a command-line tool written in V that helps to maintain a local copy of all git repositories on GitHub and Gitea.
+If you want to install it, you have to clone the source yourself, run `v build.vsh` and then add the `bin` folder to `PATH`.
+A nice thing to have is a `v fetch <repo>` or `v obtain <repo>` command (or something else, please comment) that will clone the repo and build it for you.
+
+Only once, like when V is install, for example, the user needs to add `~/.v/bin` to `PATH`.
+This will make all third-party applications installed using `v fetch <repo>` available from anywhere.
```

Comment:
This is a good point, and I think that each time we use a package manager we should create `~/.v/bin` directory. with the following rules.

 `~/.v/`
- `packages/`: all the sources downloaded
- `bin/`: all the binary downloaded, like if you want to use a V application, you can run `v install xxxxx`  and has the app up and running in the cmd
- `auth/`: some authentication file? maybe authentication if we want push v packages?
- `cache/`: store all the information useful for the bug reports, and more general v debugging

Reply:
ps: the name are randomly chosen, it is only to give the idea

Reply:
Yes, this is the exact reason I want us to move to the `~/.v` directory and sub-directory approach.